perllib/FixMyStreet/Roles/EnforcePhotoSizeOpen311PreSend.pm

```diff
@@ -0,0 +1,57 @@
+package FixMyStreet::Roles::EnforcePhotoSizeOpen311PreSend;
+use Moo::Role;
+
+=head1 NAME
+
+FixMyStreet::Roles::EnforcePhotoSizeOpen311PreSend - limit report photo sizes on open311 pre-send
+
+=head1 SYNOPSIS
+
+Applied to a cobrand class to shrink any images larger than a given size as an open311 pre-send action.
+
+Oversized images are repeatedly shrunk until they conform.
+
+A 'photo_size_limit_applied_<bytes>' metadata flag is set on the report to indicate it has been processed
+and prevent reprocessing.
+
+=cut
+
+=head1 REQUIRED METHODS
+
+=cut
+
+=head2 per_photo_size_limit_for_report_in_bytes
+
+Takes the report and the number of images.
+Returns the max number of bytes for each photo on the report.
+0 indicates no max to apply.
+
+=cut
+
+requires 'per_photo_size_limit_for_report_in_bytes';
+
+sub open311_pre_send { }
```

Comment:
I assume this is needed because otherwise the after does not fire if there isn't one in the class? Is fine, just wondering if there's a better way to structure things at a future point.

Reply:
Yeah, `after open311_pre_send` will fail to compile if the method isn't somewhere in the class hierarchy.